pkg/autoscaler/autoscaler.go

```diff
@@ -429,15 +436,62 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
 			return true
 		}, framework.WaitMedium, pollingInterval).Should(BeZero())
 
+		// Wait until the MachineSets have reach minimum size so that
+		// we can test to ensure that the remaining Machines and Nodes
+		// do not contain deletion information.
+		By("Waiting for transient machinesets to reach minimum size")
+		transientMachineSetSelector := &metav1.LabelSelector{
+			MatchLabels: map[string]string{machinesetTestLabel: ""},
+		}
+		Eventually(func() bool {
+			transientMachineSets, err := framework.GetMachineSets(client, transientMachineSetSelector)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
This would be better like this, so that we try the GetMachineSets over and over if it fails, until the timeout. Ginkgo will assert any parameters other than the first are nil when performing the `Should`, so this is safe to do
```suggestion
		Eventually(func() (bool, error) {
			transientMachineSets, err := framework.GetMachineSets(client, transientMachineSetSelector)
			if err != nil {
			    return false, err
			}
```

Reply:
that makes sense to me. i copied this logic from other places in the code and i figured if we created an error on these calls something is seriously wrong, but i can see the wisdom in just having it retry. i'll update.